src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -333,21 +333,26 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, BindEntry bindE
             var changeExpressionTokens = new List<IntermediateToken>();
 
             // DOM-based event handlers use EventCallback, so if we see a delegate here it's a component.
-            if (changeAttribute != null && changeAttribute.IsDelegateProperty())
+            if (parent is ComponentIntermediateNode && changeAttribute != null && changeAttribute.IsDelegateProperty())
             {
                 RewriteNodesForComponentDelegateBind(
                     original,
                     valueExpressionTokens,
                     changeExpressionTokens);
             }
+            else if (parent is ComponentIntermediateNode)
+            {
+                RewriteNodesForComponentEventCallbackBind(
+                    original,
+                    valueExpressionTokens,
+                    changeExpressionTokens);
+            }
             else
             {
-                RewriteNodesForEventCallbackBind(
+                RewriteNodesForElementEventCallbackBind(
```

Comment:
Could you add some comments to each of these if blocks to give context as to what scenarios can result in each of these paths being hit. Without a ton of context I'm having a hard time deciphering how/when each of these happen

Reply:
Sure!